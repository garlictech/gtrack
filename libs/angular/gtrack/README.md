# angular-gtrack

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test angular-gtrack` to execute the unit tests.
