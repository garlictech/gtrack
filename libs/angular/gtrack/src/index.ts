export * from './lib/angular-gtrack.module';
