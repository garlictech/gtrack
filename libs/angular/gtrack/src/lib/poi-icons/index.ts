export { PoiIconRowComponentModule } from './components/poi-icon-row.component';
export { PoiIconComponentModule } from './components/poi-icon.component';
export { PoiIconLabelledComponentModule } from './components/poi-icon-labelled.component';
export { PoiIconRowLabelledComponentModule } from './components/poi-icon-row-labelled.component';
