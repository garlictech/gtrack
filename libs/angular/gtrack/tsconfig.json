{
  "extends": "../../../tsconfig.base.json",
  "angularCompilerOptions": {
    "strictInjectionParameters": false,
    "strictTemplates": false
  },
  "compilerOptions": {
    "strictPropertyInitialization": false,
    "noImplicitAny": false,
    "strict": false
  },
  "files": [],
  "include": [],
  "references": [
    {
      "path": "./tsconfig.lib.json"
    },
    {
      "path": "./tsconfig.spec.json"
    }
  ]
}