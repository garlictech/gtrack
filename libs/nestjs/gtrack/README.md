# nestjs-gtrack

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `ng test nestjs-gtrack` to execute the unit tests via [Jest](https://jestjs.io).
