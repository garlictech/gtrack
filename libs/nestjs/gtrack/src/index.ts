export * from './lib/nestjs-gtrack.module';
