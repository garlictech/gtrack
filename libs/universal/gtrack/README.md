# universal-gtrack

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `ng test universal-gtrack` to execute the unit tests via [Jest](https://jestjs.io).
