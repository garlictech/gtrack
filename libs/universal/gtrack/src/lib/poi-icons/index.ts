export * from './lib/poi-icons.fp';
export * from './lib/types';
export * from './lib/supported-tags';
