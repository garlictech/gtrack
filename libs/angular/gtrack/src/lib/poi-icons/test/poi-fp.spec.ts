import { PoiIconsFp } from '@bit/garlictech.universal.gtrack.poi-icons';

const fixture1 = 'amenity:casino';
const fixture2 = 'natural:wood';

test('PoiIconsFp.getIcondescriptor must work', () => {
  expect(PoiIconsFp.getIcon(fixture1)).toMatchSnapshot('1');
  expect(PoiIconsFp.getIcon('')).toMatchSnapshot('2');
  expect(PoiIconsFp.getIcon('amenity:')).toMatchSnapshot('3');
  expect(PoiIconsFp.getIcon(':wood')).toMatchSnapshot('4');
  expect(PoiIconsFp.getIcon(':')).toMatchSnapshot('5');
  expect(PoiIconsFp.getIcon('amenity')).toMatchSnapshot('6');
  expect(PoiIconsFp.getIcons([fixture1, fixture2])).toMatchSnapshot('7');
  expect(PoiIconsFp.getIcons([fixture1, ''])).toMatchSnapshot('8');
});
