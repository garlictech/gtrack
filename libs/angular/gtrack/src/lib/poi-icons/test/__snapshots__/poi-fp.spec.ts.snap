// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PoiIconsFp.getIcondescriptor must work: 1 1`] = `
Object {
  "_tag": "Some",
  "value": Object {
    "color": "#265DB2",
    "url": "amenity:casino",
  },
}
`;

exports[`PoiIconsFp.getIcondescriptor must work: 2 1`] = `
Object {
  "_tag": "None",
}
`;

exports[`PoiIconsFp.getIcondescriptor must work: 3 1`] = `
Object {
  "_tag": "None",
}
`;

exports[`PoiIconsFp.getIcondescriptor must work: 4 1`] = `
Object {
  "_tag": "None",
}
`;

exports[`PoiIconsFp.getIcondescriptor must work: 5 1`] = `
Object {
  "_tag": "None",
}
`;

exports[`PoiIconsFp.getIcondescriptor must work: 6 1`] = `
Object {
  "_tag": "None",
}
`;

exports[`PoiIconsFp.getIcondescriptor must work: 7 1`] = `
Object {
  "_tag": "Some",
  "value": Array [
    Object {
      "color": "#265DB2",
      "url": "amenity:casino",
    },
    Object {
      "color": "#108E4C",
      "url": "generic:asterisco",
    },
  ],
}
`;

exports[`PoiIconsFp.getIcondescriptor must work: 8 1`] = `
Object {
  "_tag": "None",
}
`;
